packages/cli/src/commands/env/pull.ts

```diff
@@ -135,3 +152,79 @@ function escapeValue(value: string | undefined) {
         .replace(new RegExp('\r', 'g'), '\\r') // combine newlines (windows) into one line
     : '';
 }
+
+async function createEnvObject(
+  envPath: string,
+  output: Output
+): Promise<Dictionary<string | undefined> | undefined> {
+  try {
+    // Inspired by https://github.com/tswaters/env-file-parser/blob/master/lib/parse.js
+    let envArr = (await readFile(envPath, 'utf-8'))
+      // remove double quotes
+      .replace(/"/g, '')
+      // split on new line
+      .split(/\r?\n|\r/)
+      // filter comments
+      .filter(line => /^[^#]/.test(line))
+      // needs equal sign
+      .filter(line => /=/i.test(line));
+    return parseEnv(envArr);
+  } catch (err) {
+    output.debug(`Error parsing env file: ${err}`);
+  }
+}
+
+function findChanges(
+  oldEnv: Dictionary<string | undefined>,
+  newEnv: Dictionary<string | undefined>
+): {
+  added: string[];
+  changed: string[];
+  removed: string[];
+} {
+  let added = [];
+  let changed = [];
+  let removed = [];
+
+  for (const key of Object.keys(newEnv)) {
+    if (!oldEnv[key] && oldEnv[key] !== '') {
+      added.push(key);
+    } else if (oldEnv[key] !== newEnv[key]) {
+      changed.push(key);
+    }
+    delete oldEnv[key];
+  }
+  removed = Object.keys(oldEnv);
+
+  return {
+    added,
+    changed,
+    removed,
+  };
+}
+
+function buildDeltaString(
+  oldEnv: Dictionary<string | undefined>,
+  newEnv: Dictionary<string | undefined>
+): string {
+  let { added, changed, removed } = findChanges(oldEnv, newEnv);
+
+  let deltaString = '';
+  deltaString += chalk.green(addDeltaSection('+', added));
+  deltaString += chalk.yellow(addDeltaSection('~', changed));
```

Comment:
I think it makes sense that `~` would mean "changed", but not immediately. I think a user may not know what this means, especially if their first interactions with this output only shows one variable changed.

Should we log a `(changed)` suffix (maybe dimmed) at the end of these lines? Or can we show a brief legend without taking up too much spaced?

Reply:
I'll explore a legend, though not quite sure yet how to make it look good & not feel awkward. Big fan of the `(changed)` suffix.

Reply:
Added a `Changed:` prefix, but not super confident in this

Reply:
I think the `Changed:` prefix is fine, but not necessary. I think the issue with `~` still stands, though.

Reply:
Thinking about it more, I'm not convinced this is an issue. I think there is enough context as to what the tilde means: there is now a "changed" prefix which gives a clue, especially if there's only one edited variable, and the yellow color indicates (at least to me) that the variable wasn't added or removed, but somewhere in between—AKA edited.

I think a legend, or more explicit context, will take up a lot of the space, and is the sort of thing that may be useful to see once but will be annoying to see every time after that.

I would be happy to ship this as is. Maybe we should bring this to the rest of the team or to `#design` / `#engineering` for additional feedback.